cmake/external/ngraph.cmake

```diff
@@ -0,0 +1,138 @@
+    MESSAGE(STATUS "Configure nGraph with MKLDNN: INCLDUE ${MKLDNN_INCLUDE_DIR}, LIB ${MKLDNN_LIB_DIR}")
+    ExternalProject_Add(
+        ${NGRAPH_PROJECT}
+        ${EXTERNAL_PROJECT_LOG_ARGS}
+        DEPENDS             ${NGRAPH_DEPENDS}
+        GIT_REPOSITORY      ${NGRAPH_GIT_REPO}
+        GIT_TAG             ${NGRAPH_GIT_TAG}
+        PREFIX              ${NGRAPH_SOURCES_DIR}
+        UPDATE_COMMAND      ""
+        CMAKE_ARGS          -DNGRAPH_INSTALL_PREFIX=${NGRAPH_INSTALL_DIR}
+        # CMAKE_ARGS          -DNGRAPH_DEX_ONLY=TRUE
+        CMAKE_ARGS          -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        CMAKE_ARGS          -DNGRAPH_USE_PREBUILT_LLVM=${NGRAPH_USE_PREBUILT_LLVM}
+        CMAKE_ARGS          -DMKLDNN_INCLUDE_DIR=${MKLDNN_INCLUDE_DIR}
+        CMAKE_ARGS          -DMKLDNN_LIB_DIR=${MKLDNN_LIB_DIR}
+        CMAKE_CACHE_ARGS    -DNGRAPH_INSTALL_PREFIX:PATH=${NGRAPH_INSTALL_DIR}
+    )
+    ExternalProject_Add_Step(
+        ${NGRAPH_PROJECT}
+        PREDOWNLOAD
+        DEPENDERS download
+        COMMAND ${CMAKE_COMMAND} -E copy_directory ${MKLML_ROOT}/lib ${MKLDNN_LIB_DIR}
+        )
+
+ELSE()
+    ExternalProject_Add(
+        ${NGRAPH_PROJECT}
+        ${EXTERNAL_PROJECT_LOG_ARGS}
+        GIT_REPOSITORY      ${NGRAPH_GIT_REPO}
+        GIT_TAG             ${NGRAPH_GIT_TAG}
+        PREFIX              ${NGRAPH_SOURCES_DIR}
+        UPDATE_COMMAND      ""
+        CMAKE_ARGS          -DNGRAPH_INSTALL_PREFIX=${NGRAPH_INSTALL_DIR}
+        # CMAKE_ARGS          -DNGRAPH_DEX_ONLY=TRUE
+        CMAKE_ARGS          -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        CMAKE_ARGS          -DNGRAPH_USE_PREBUILT_LLVM=${NGRAPH_USE_PREBUILT_LLVM}
+        CMAKE_CACHE_ARGS    -DNGRAPH_INSTALL_PREFIX:PATH=${NGRAPH_INSTALL_DIR}
+    )
+ENDIF()
+
+ADD_LIBRARY(shared_ngraph SHARED IMPORTED GLOBAL)
+SET_PROPERTY(TARGET shared_ngraph PROPERTY IMPORTED_LOCATION ${NGRAPH_LIB})
+ADD_DEPENDENCIES(shared_ngraph ${NGRAPH_PROJECT})
+MESSAGE(STATUS "nGraph library: ${NGRAPH_LIB}")
+add_definitions(-DPADDLE_WITH_NGRAPH)
+LIST(APPEND external_project_dependencies shared_ngraph)
+
+# generate a static dummy target to track nGraph dependencies
+# for cc_library(xxx SRCS xxx.c DEPS ngraph)
+SET(dummyfile ${CMAKE_CURRENT_BINARY_DIR}/ngraph_dummy.c)
+FILE(WRITE ${dummyfile} "const char * dummy = \"${dummyfile}\";")
+ADD_LIBRARY(ngraph STATIC ${dummyfile})
+TARGET_LINK_LIBRARIES(ngraph ${NGRAPH_LIB} ${MKLDNN_LIB} ${MKLML_LIB} ${MKLML_IOMP_LIB})
+ADD_DEPENDENCIES(ngraph ${NGRAPH_PROJECT})
+
+# copy the real so.0 lib to install dir
+# it can be directly contained in wheel or capi
+SET(NGRAPH_SHARED_LIB ${NGRAPH_INSTALL_DIR}/libngraph.so.0.8)
+ADD_CUSTOM_COMMAND(OUTPUT ${NGRAPH_SHARED_LIB}
+    COMMAND cp ${NGRAPH_LIB} ${NGRAPH_SHARED_LIB}
+    DEPENDS ngraph)
+ADD_CUSTOM_TARGET(ngraph_shared_lib ALL DEPENDS ${NGRAPH_SHARED_LIB})
+
+IF(WITH_C_API)
+    INSTALL(FILES ${NGRAPH_SHARED_LIB} DESTINATION lib)
+ENDIF()
```

Comment:
remove line 135-167, since CAPI is paddle v2 version.

Reply:
will address in the separate cmake PR